components/server/ee/src/prebuilds/incremental-prebuilds-service.ts

```diff
@@ -68,12 +68,15 @@ export class IncrementalPrebuildsService {
         context: CommitContext,
         history: WithCommitHistory,
         user: User,
+        config?: WorkspaceConfig,
     ): Promise<PrebuiltWorkspace | undefined> {
         if (!history.commitHistory || history.commitHistory.length < 1) {
             return;
         }
 
-        const { config } = await this.configProvider.fetchConfig({}, user, context);
+        if (!config) {
+            config = (await this.configProvider.fetchConfig({}, user, context)).config;
+        }
```

Comment:
We're always gonna fetch the config when it's not provided. Here, we can push that into caller and always require a config to present on this call. The caller can do `await this.configProvider.fetchConfig({}, user, context)).config` to get it. It has the benefit of not hiding complexity and the method signature much better communicates the behaviour.

We only have another caller from `gitpod-server-impl` so it is a small change.

```
config: WorkspaceConfig,
    ): Promise<PrebuiltWorkspace | undefined> {
        if (!history.commitHistory || history.commitHistory.length < 1) {
            return;
        }
```

Reply:
This implies injecting a `ConfigProvider` into `GitpodServerImplEE`, but I think that's okay and cleaner this way.